### PR TITLE
ISSUE-31-33-36-41: fix several issues

### PR DESCRIPTION
**Pull request by qiawu**
This PR includes:
1. make test more extensible (ISSUE-31)
2. change grpc message limit (ISSUE-33)
3. expose cluster info from raft layer (ISSUE-36)
4. support dns resolver when reconnect (ISSUE-41)

### CODE REVIEW

File: src/infra/raft/v2/RaftService.cpp [hunk 1]
```diff
@@ -51,6 +51,7 @@ void RaftServer::serverLoopMain() {
   pthread_setname_np(pthread_self(), "RaftServer");
 
   grpc::ServerBuilder builder;
+  builder.SetMaxReceiveMessageSize(INT_MAX);
   builder.AddListeningPort(mIpPort, TlsUtil::buildServerCredentials(mTlsConfOpt));
```
Comment (lingjun17) [inline src/infra/raft/v2/RaftService.cpp:55]: const auto &ipaddr = mDNSResolver.resolve(mIpPort);
RaftCore mutual tls is enabled, raftserver also needs enable parser.

File: src/infra/raft/v2/RaftService.cpp [hunk 2]
```diff
@@ -86,9 +87,17 @@ void RaftServer::serverLoopMain() {
...
+                       std::optional<TlsConf> tlsConfOpt,
+                       std::shared_ptr<DNSResolver> dnsResolver,
+                       uint64_t peerId,
+                       EventQueue *aeRvQueue) :
+    mPeerAddress(peerAddress),
+    mDNSResolver(dnsResolver),
+    mTLSConfOpt(tlsConfOpt),
+    mPeerId(peerId),
+    mAeRvQueue(aeRvQueue) {
+  refressChannel();
```
Comment (lingjun17) [inline src/infra/raft/v2/RaftService.cpp:100]: maybe we needn't to refresh all clients, only refreshing the connection error nodes is enough.

File: src/infra/raft/v2/RaftService.cpp (hunk 1 above)
Comment (qiawu) [reply src/infra/raft/v2/RaftService.cpp:55]: not an issue

File: src/infra/raft/v2/RaftService.cpp (hunk 2 above)
Comment (qiawu) [reply src/infra/raft/v2/RaftService.cpp:100]: already did in refresh logic

**Review by lingjun17 — APPROVED** [review]
LGTM

**Review by lingjun17 — APPROVED** [review]

**Review by huyumars — APPROVED** [review]

**Review by huyumars — DISMISSED** [review]

**Review by huyumars — APPROVED** [review]

**Pull request merged**